Code-the-Dream-School/gg-pac-team3-monorepo
# quiz, user course and main tests

**Testing Files:**

- quizController.tests.mjs: Tests for quiz creation, fetching, updating, and deletion.

- userCourseController.tests.mjs: enrolling a user, fetching, and unknown user.

- mainController.tests.mjs: verify the data response.

- Refactor the testing to separate the backend and frontend tests

Testing Instructions:

1. I updated the package.json to have the option to test each file.

2. Before running the tests make sure you have all dependencies.

3. To run all the tests:
`  npm run dev `
 `npm test`

**4. To do individual testings:**

To run the user controller tests:
`  npm run test:user`

To run the frontpage tests:
`npm run dev`
`npm run test:frontpage`

To run the course controller tests:
`npm run test:course`

To run the lesson controller tests:
`npm run test:lesson`

To run the quiz controller tests:
`npm run test:quiz`

To run the user course controller tests:
`npm run test:userCourse`
<img width="579" alt="Screenshot 2024-08-25 at 6 44 08 PM" src="https://github.com/user-attachments/assets/33d63641-6a96-4937-b047-f5e61f24b67b">
